libs/clients/social-insurance-administration/src/lib/socialInsuranceAdministrationClient.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Auth, AuthMiddleware, User } from '@island.is/auth-nest-tools'
+import { Injectable } from '@nestjs/common'
+import {
+  OldAgePension,
+  SendApplicationApi,
+  Response,
+  GetIsApplicantEligibleApi,
+  Eligible,
+  GetApplicantInfoApi,
+  Applicant,
+  GetCurrenciesApi,
+} from '../../gen/fetch'
+
+@Injectable()
+export class SocialInsuranceAdministrationClientService {
+  constructor(
+    private readonly sendApplicationApi: SendApplicationApi,
+    private readonly getIsApplicantEligibleApi: GetIsApplicantEligibleApi,
+    private readonly getApplicantInfoApi: GetApplicantInfoApi,
+    private readonly getCurrenciesApi: GetCurrenciesApi,
+  ) {}
+
+  private sendAPIWithAuth = (user: User) =>
+    this.sendApplicationApi.withMiddleware(new AuthMiddleware(user as Auth))
+  private isEligibleAPIWithAuth = (user: User) =>
+    this.getIsApplicantEligibleApi.withMiddleware(
+      new AuthMiddleware(user as Auth),
+    )
+  private applicantAPIWithAuth = (user: User) =>
+    this.getApplicantInfoApi.withMiddleware(new AuthMiddleware(user as Auth))
+  private currenciesAPIWithAuth = (user: User) =>
+    this.getCurrenciesApi.withMiddleware(new AuthMiddleware(user as Auth))
+
+  async sendApplication(
+    user: User,
+    oldAgePension: OldAgePension,
+    applicationType: string,
+  ): Promise<Response> {
+    return await this.sendAPIWithAuth(user).oldAgePensionSendApplication({
+      oldAgePension,
+      applicationType,
+    })
+  }
```

Comment:
~ When just returning the promise from the api call `async/await` isn't required. `return await` is only required inside a `try/catch` when we want to wait for the response to let the catch handle the error.
```suggestion
  sendApplication(
    user: User,
    oldAgePension: OldAgePension,
    applicationType: string,
  ): Promise<Response> {
    return this.sendAPIWithAuth(user).oldAgePensionSendApplication({
      oldAgePension,
      applicationType,
    })
  }
```